lib/vrl/stdlib/src/unnest.rs

```diff
@@ -327,9 +378,45 @@ mod tests {
                     } },
                 ] },
             },
-            // Coalesce
+            //// Coalesce with known path first
+            ////
+            //// FIXME(Jean): There's still a bug in the `InsertValid` implementation that prevents
+            //// this from working as expected.
```

Comment:
I'm leaving this test commented out. I've filed an issue at #11249 to follow up on this. To be clear, this is existing behaviour _and_ it's not _incorrect_ information given by the type checker, but it's _incomplete_, meaning it's result is more expansive than it could be.